### ISSUE TITLE
Add Opacity & Variable Height to Minimap

### ISSUE THREAD

**Issue by IMBurbank** [Microsoft/vscode#32581]
I'm looking for a way to change my VSCode Editor Minimap to behave more like the Atom Minimap.

Specifically, to have container opacity and variable height as the file grows in length. The opacity is what is most important to me, but the variable container length would help a lot as well.

Currently, my Atom minimap allows opacity and only grows in length as needed.  
![capture](https://user-images.githubusercontent.com/21133719/29327005-e4b47bee-81aa-11e7-8cd9-52cb6411d8c0.JPG)

While my VSCode minimap covers my code and maintains full pane height no matter how many lines are in the file. 
![capture1](https://user-images.githubusercontent.com/21133719/29327030-f82e16b2-81aa-11e7-8107-07972643f65f.JPG)

I tried asking a question about this on [stackoverflow](https://stackoverflow.com/questions/45685834/vscode-minimap-creating-opacitylimited-height-like-atom-minimap-package) and looking through these issues. So far the closest I've found to this question is issue [#21255](https://github.com/Microsoft/vscode/issues/21255), which points to issue [#21229](https://github.com/Microsoft/vscode/issues/21229) and then to issue [#21359](https://github.com/Microsoft/vscode/issues/21359), which lead to the implementation of a box shadow to make the minimap separation more prominent.

Unfortunately, the box shadow doesn't allow me to recover that text editor width while keeping my minimap enabled. This makes the difference between being able to see all my code using 2 panes and having my code cut off by the minimap.

I'm sure a lot of people transitioning from Atom to VSCode would appreciate being able to maintain this editor workspace, particularly when working with a lot of smaller files.

**Comment by IMBurbank** [comment]
I know this is a big project and don't want to bug you guys - I'm not sure how long is standard for a response and don't want to jump the gun - but as it's been a week I have to ask: is there anything I can do to aid in a decision or implementation? Or help this along generally?

Unfortunately, I have no experience contributing to open-source projects. But I am an engineer, and I've been teaching myself to code for the last year. I'm just starting to approach a low level of usefulness in my coding endeavors.

**Comment by nealot** [comment]
Variable height might be related to https://github.com/Microsoft/vscode/issues/21773. 

I find Atom's minimap actually useful as it clearly outlines the structure of a file, whereas in vscode the minimap needs me to strain my eyes to view the same content.

While everything else is scalable, you can't modify the size of the minimap at all. That means people who suffer from poor eyesight or those who have hdpi screens just have to accept poor accessibility.

**Comment by kokopelli314** [comment]
I've started doing a bit of work on this issue. When the minimap's background is transparent, I think the variable height could take care of itself, as in this screenshot:

![image](https://user-images.githubusercontent.com/10605105/46927476-da853b80-cff2-11e8-9eda-70a6f235b8ed.png)

The minimap height in this short file _seems_ short, because only the editor is visible where the minimap has no code. (In this example, I've also set the `scrollbar.shadow` option in settings.json to transparent.)

So far, I'm looking at adding settings.json options for `minimap.backgroundAlpha` and `minimap.fontAlpha`, so that the text/block's transparency can be controlled as well the background's. Definitely open to ideas on how to what would be most useful in controlling the settings!

**Comment by gloss-water** [comment]
Hi @kokopelli314 !  I semi-recently PR'd a similar feature to this as well -- #58324 -- and the discussion has died off there with the UI clash being dismissed outright.  This is a feature that I'd really like to see, whether it's my personal contribution or not, and I'm commenting here mostly in the hopes that it helps this feature to not get overlooked :)

I added some additional comments in that original PR, but I'll reiterate here that this is a feature available in other editors already, specifically with Atom's minimap opacity options and Sublime Text's ability to override the class names and their opacity values.

**Comment by kokopelli314** [comment]
@gloss-water Thanks for sharing! The option enabled in your PR looks good to me as well. 

I agree this would be highly useful, and other editors I've used (especially Atom) already have this as a setting.

**Comment by RJSonnenberg** [comment]
@alexandrudima When is this going to be merged? Looks like the PR is just waiting. This would be an awesome feature that I would use right away!

**Comment by tommyalatalo** [comment]
Also waiting for this one, please merge if PR is done :)

**Comment by dmmello1989** [comment]
Any news on this? 
The minimap blocking my lines is really bugging me, I'd love some opacity settings to be available :)

**Comment by opensas** [comment]
Hate to be one of those 'me too' guys, but I'm also waiting for this to be merged. Minimap is prety useful WHEN yo need it, but with the current implementation it always ends up eating precious screen space, setting the opacity level would be great, on mouse over it could come back without opacity...

**Comment by gloss-water** [comment]
@opensas at this point the feature has been PR'd multiple times over the last couple years, with varying methods of implementing it.  the PRs get ignored each time, the [only real response i've seen so far](https://github.com/microsoft/vscode/pull/58324#issuecomment-420343521) isn't particularly promising or helpful for moving forward either

**Comment by IMBurbank** [comment]
This feature is just as important today as it was 2.5 years ago.

It's quite disappointing that this feature request has never been directly addressed, even in response to PRs, particularly since it's an opt-in feature - minimap opacity would only change if set to less than 100%.

If there's a reason this feature request won't be accepted, I think a lot of people would appreciate knowing that so they can act accordingly.
If this feature would be accepted in some form, I would be more than happy to submit a new PR for it.

**Comment by jgburet** [comment]
up 💃

**Comment by LucasBonafe** [comment]
### Any solution?

**Comment by jdr-face** [comment]
Yes. See: https://github.com/Microsoft/vscode/pull/61021#issuecomment-582520247

**Comment by robinmacharg** [comment]
To save a click-through for anyone landing here, CMD-SHIFT-P -> Open Settings (JSON), and paste the following into the top level of the JSON:

```JSON
"workbench.colorCustomizations": {
    "minimap.background": "#ffffff33"
}
```

This fixes the opacity (I use #000000 instead of white).  I can live without the variable height.

**Comment by jaquinocode** [comment]
Try `"#1e1e1e33"` which is the exact color of vscode's default theme background (at least on my mac) which makes it look a lot nicer.

For those who don't know, this is a hexadecimal format for red, green, blue, and opacity. Last two digits controls the opacity.

**Comment by hlizard** [comment]
"#4b384d5e" with SynthWave '84

**Comment by MaltesBytes** [comment]
**_### Any solution?_**

**Comment by dailytabs** [comment]
Really don't like clicking in from another window and being taken to the bottom of the file unexpectedly, because the minimap extends invisibly to the bottom of the editor. This is not expected nor intuitive.
